### ISSUE TITLE
Value cast to TypeVar generates false positive error

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4218]
### Discussed in https://github.com/microsoft/pyright/discussions/4217

<div type='discussions-op-text'>

<sup>Originally posted by **alxrcs** November 21, 2022</sup>
Assume the following example:

```python
from typing import TypeVar, Type, cast, Generic
from copy import copy

T = TypeVar("T", bound="Castable")

class Castable(Generic[T]):
    t: Type[T]

    def to_t(self) -> T:
        c = copy(self)
        c.__class__ = self.t
        return cast(T, c)

    def __init__(self, t: Type[T]):
        self.t = t
```

pyright 1.1.280 shows a somewhat confusing error of `Type "T@Castable" cannot be assigned to type "T@Castable"`
mypy 0.991 doesn't complain about this.

This should typecheck, right? Or is there something I'm missing and it shouldn't? </div>

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.281, which I just published. It will also be included in a future release of pylance.

**Comment by alxrcs** [comment]
That was fast, thanks a bunch!
